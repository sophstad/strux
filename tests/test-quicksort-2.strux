int main() {
	int[] a = [1, 4, 3, 6, 7, 2, 99, 23, 37];
  a.showQuickSort(9);
	return 0;
}
