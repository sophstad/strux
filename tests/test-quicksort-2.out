=====================================================================================
At this step:
current array: [ 1 4 3 6 7 2 99 23 37 ]
numbers swapped: 23,6
array after swap: [ 1 4 3 23 7 2 99 6 37 ]
pivot is 6
numbers swapped: 99,6
array after swap: [ 1 4 3 23 7 2 6 99 37 ]
numbers swapped: 37,99
array after swap: [ 1 4 3 23 7 2 6 37 99 ]
=====================================================================================
At this step:
current array: [ 1 4 3 23 7 2 6 37 99 ]
numbers swapped: 2,3
array after swap: [ 1 4 2 23 7 3 6 37 99 ]
pivot is 3
numbers swapped: 23,3
array after swap: [ 1 4 2 3 7 23 6 37 99 ]
numbers swapped: 6,7
array after swap: [ 1 4 2 3 6 23 7 37 99 ]
=====================================================================================
At this step:
current array: [ 1 4 2 3 6 23 7 37 99 ]
pivot swapped: 4,3
array after swap: [ 1 3 2 4 6 23 7 37 99 ]
numbers swapped: 2,3
array after swap: [ 1 2 3 4 6 23 7 37 99 ]
pivot is 3
=====================================================================================
At this step:
current array: [ 1 2 3 4 6 23 7 37 99 ]
numbers swapped: 2,1
array after swap: [ 2 1 3 4 6 23 7 37 99 ]
pivot is 1
=====================================================================================
At this step:
current array: [ 2 1 3 4 6 23 7 37 99 ]
pivot swapped: 2,1
array after swap: [ 1 2 3 4 6 23 7 37 99 ]
pivot is 1
=====================================================================================
At this step:
current array: [ 1 2 3 4 6 23 7 37 99 ]
pivot is 0
=====================================================================================
At this step:
current array: [ 1 2 3 4 6 23 7 37 99 ]
pivot swapped: 23,7
array after swap: [ 1 2 3 4 6 7 23 37 99 ]
pivot is 7
=====================================================================================
QuickSort complete! Final Result: [ 1 2 3 4 6 7 23 37 99 ]
